substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -33,16 +33,207 @@ use codec::{Compact, Decode, Encode, EncodeLike, Error, Input};
 use core::fmt;
 use scale_info::{build::Fields, meta_type, Path, StaticTypeInfo, Type, TypeInfo, TypeParameter};
 use sp_io::hashing::blake2_256;
+use sp_weights::Weight;
+
+/// Type to represent the version of the [Extension](TransactionExtension) used in this extrinsic.
+pub type ExtensionVersion = u8;
+/// Type to represent the extrinsic format version which defines an [UncheckedExtrinsic].
+pub type ExtrinsicVersion = u8;
```

Comment:
These two versions should go to the metadata (I propose `Version` type, found at `System.Version` constant).

Reply:
The extrinsic version (soon to be a vec of supported versions in V16 metadata) is already in there.

The extension version we should add in some way. I believe it changes when the set of transaction extensions in a runtime changes? In which case we either want to add one extension version to the metadata to go with the list of "current" transaction extensions, or we could to have a vec of `(extension_version, list_of_extensions)` to reflect the current sets of extensions that the runtime supports and the version of each, if applicable.

Reply:
Probably adding it to `System.Version` is the way to proceed.

Reply:
@gui1117 @georgepisaltu thoughts here?

Reply:
> The extension version we should add in some way. I believe it changes when the set of transaction extensions in a runtime changes?

The extension version here doesn't change if a runtime add or remove some transaction extensions.
It is 0 for all runtime currently.
If a transaction extension get added or removed, I think we should bump the transaction version in the runtime version (`RuntimeVersion` struct, `transaction_version` field), because it changes the execution of the extrinsic.

I don't exactly know when we would bump it but IIUC extension version is associated with the way transaction extensions are used in an extrinsic, in the implementation in sp-runtime.

Reply:
[RFC99](https://github.com/polkadot-fellows/RFCs/blob/main/text/0099-transaction-extension-version.md) defines the transaction extension version. This PR only adds encoding/decoding support for this to not break the format twice (as is menitoned in the RFC as well). Full support for the extension version is out of scope for this PR and will be added later. Metadata work is also a follow-up on this PR.

Reply:
I still think the extension version should be part of the inherited implication, so it can be signed in some way.
Otherwise supported multiple version can lead to extrinsic being valid in some non-intended context.
See : https://github.com/paritytech/polkadot-sdk/pull/3685#issuecomment-2400404179

Reply:
Replied [here](https://github.com/paritytech/polkadot-sdk/pull/3685#issuecomment-2400586757) and I think it has merit but I stand by my earlier comment above about this being out of scope for this PR.